langchain/callbacks/whylabs_callback.py

```diff
@@ -0,0 +1,161 @@
+import logging
+from typing import Any, Dict, List, Optional, Union
+
+from langchain.callbacks.base import BaseCallbackHandler
+from langchain.schema import AgentAction, AgentFinish, Generation, LLMResult
+
+logger = logging.getLogger(__name__)
+
+
+def import_langkit(
+    sentiment: Optional[bool] = None,
+    toxicity: Optional[bool] = None,
+    themes: Optional[bool] = None,
+) -> Any:
+    try:
+        import langkit  # noqa: F401
+        import langkit.regexes  # noqa: F401
+        import langkit.textstat  # noqa: F401
+
+        if sentiment:
+            import langkit.sentiment  # noqa: F401
+        if toxicity:
+            import langkit.toxicity  # noqa: F401
+        if themes:
+            import langkit.themes  # noqa: F401
+    except ImportError:
+        raise ImportError(
+            "To use the whylabs callback manager you need to have the `langkit` python "
+            "package installed. Please install it with `pip install langkit`."
+        )
+    return langkit
+
+
+class WhyLabsCallbackHandler(BaseCallbackHandler):
+    """WhyLabs CallbackHandler."""
+
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        org_id: Optional[str] = None,
+        dataset_id: Optional[str] = None,
+        sentiment: Optional[bool] = None,
+        toxicity: Optional[bool] = None,
+        themes: Optional[bool] = None,
+    ):
+        """Initiate the rolling logger"""
+        super().__init__()
+
+        # langkit library will import necessary whylogs libraries
+        import_langkit(sentiment=sentiment, toxicity=toxicity, themes=themes)
+
+        import whylogs as why
+        from whylogs.api.writer.whylabs import WhyLabsWriter
+        from whylogs.core.schema import DeclarativeSchema
+        from whylogs.experimental.core.metrics.udf_metric import generate_udf_schema
+
+        self.writer = WhyLabsWriter(
```

Comment:
do we need to store this or is it only passed in to logger?

Reply:
We don't need to store it with the current integration, lets simplify and not have direct access to the writer yet, removing.